### PR TITLE
Remove system.random 

### PR DESCRIPTION
**Pull request by cristinamanum**
Remove system.random usage.

### CODE REVIEW

File: tests/NuGetGallery.Facts/Services/TyposquattingServiceFacts.cs [hunk 1]
```diff
@@ -4,10 +4,10 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading.Tasks;
 using Moq;
 using NuGet.Services.Entities;
 using Xunit;
+using NuGetGallery.TestUtils;
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/Services/TyposquattingServiceFacts.cs]: sort usings

File: tests/NuGetGallery.Facts/Infrastructure/Authentication/Base32EncoderTests.cs [hunk 2]
```diff
@@ -6,12 +6,13 @@
...
 using NuGetGallery.Infrastructure.Authentication;
 using Xunit;
+using NuGetGallery.TestUtils;
 
 namespace NuGetGallery.Infrastructure
 {
     public class Base32EncoderTests
     {
-        private Random _random = new Random(0);
+        private SecureRandomNumberGenerator _random = new SecureRandomNumberGenerator();
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/Infrastructure/Authentication/Base32EncoderTests.cs]: Is this actually a requirement for test code?

File: tests/NuGetGallery.Facts/Infrastructure/Authentication/Base32EncoderTests.cs [hunk 3]
```diff
@@ -6,12 +6,13 @@
 using System.Text;
 using NuGetGallery.Infrastructure.Authentication;
 using Xunit;
+using NuGetGallery.TestUtils;
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/Infrastructure/Authentication/Base32EncoderTests.cs]: Sort usings

File: tests/NuGetGallery.Facts/Infrastructure/Authentication/Base32EncoderTests.cs (hunk 2 above)
Comment (scottbommarito) [reply tests/NuGetGallery.Facts/Infrastructure/Authentication/Base32EncoderTests.cs]: I agree, using secure random numbers in test code seems very arbitrary...

File: tests/NuGetGallery.Facts/TestUtils/SecureRandomNumberGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+using System;
+using System.Security.Cryptography;
+
+namespace NuGetGallery.TestUtils
+{
+    /// <summary>
+    /// A number generator using RNGCryptoServiceProvider.
+    /// Based on http://csharphelper.com/blog/2014/08/use-a-cryptographic-random-number-generator-in-c/
+    /// </summary>
+    public class SecureRandomNumberGenerator
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/TestUtils/SecureRandomNumberGenerator.cs]: This should be in a shared place if this really a new requirement for us.

Comment (joelverhagen) [reply tests/NuGetGallery.Facts/TestUtils/SecureRandomNumberGenerator.cs]: (ServerCommon)

**Pull request closed without merging**